stacker/src/expull.rs

```diff
@@ -52,50 +22,56 @@ fn len_to_capacity(len: u32) -> CapacityResult {
 /// problem of selecting an adequate block size using a strategy similar to
 /// that of the `Vec` amortized resize strategy.
 ///
-/// Data is stored in a linked list of blocks. The first block has a size of `4`
+/// Data is stored in a linked list of blocks. The first block has a size of `8`
 /// and each block has a length of twice that of the previous block up to
-/// `MAX_BLOCK_LEN = 32768`.
+/// `MAX_BLOCK_LEN = 1<<15`.
 ///
 /// This strategy is a good trade off to handle numerous very rare terms
 /// and avoid wasting half of the memory for very frequent terms.
 #[derive(Debug, Clone, Copy)]
 pub struct ExpUnrolledLinkedList {
-    len: u32,
+    // u16, since the max size of each block is (1<<next_cap_pow_2)
+    // Limited to 15, so we don't overflow remaining_cap.
+    remaining_cap: u16,
+    // To get the current number of blocks: block_num - FIRST_BLOCK_NUM
+    block_num: u16,
+    head: Addr,
     tail: Addr,
-    inlined_data: [u8; INLINED_BLOCK_LEN],
```

Comment:
before we remove this... do we have  a bench testing what happens with a unique id field?

Reply:
I added a benchmark for this case, the performance seems to be the same
```
CreateHashMap/ids_expull/8000000
                        time:   [72.704 ms 73.010 ms 73.357 ms]
                        thrpt:  [104.00 MiB/s 104.50 MiB/s 104.94 MiB/s]
                 change:
                        time:   [-0.9446% -0.3605% +0.2415%] (p = 0.24 > 0.05)
                        thrpt:  [-0.2409% +0.3618% +0.9536%]
                        No change in performance detected.
```